modules/kubeconfig/main.tf

```diff
@@ -37,8 +29,33 @@ data "template_file" "kubeconfig_sa" {
     token        = lookup(data.kubernetes_secret.sa_secret.0.data,"token", "")
     namespace    = var.namespace
   }
+  depends_on = [kubernetes_secret.sa_secret]
```

Comment:
Do you still need depends_on here after adding back the `data "kubernetes_secret" "sa_secret"`

Reply:
that should be the data element not the resource element: `data.kubernetes_server.sa_secret`

Reply:
Updated.